### PR TITLE
Page state

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Review by cnrudd — COMMENTED** [review]
The pageState logic all looks good.  And the integration into Pref service looks good too.
With so many changes to XH and ACM, are you sure there are no breaking changes?
Maybe some apps are are using XH methods that have been moved?

File: core/exception/ExceptionHandler.ts [hunk 1]
```diff
@@ -156,7 +150,8 @@ export class ExceptionHandler {
      * @param options - provides further control over how the exception is shown and/or logged.
      */
     showException(exception: unknown, options?: ExceptionHandlerOptions) {
-        if (this.#isUnloading) return;
+        if (XH.pageState == 'terminated' || XH.pageState == 'frozen') return;
```
Comment (lbwexler) [inline core/exception/ExceptionHandler.ts:153]: ===

**Comment by amcclain** [comment]
Taking a look at this today!

File: svc/IdentityService.ts [hunk 2]
```diff
@@ -69,6 +72,23 @@ export class IdentityService extends HoistService {
...
+            {checkAccess} = XH.appSpec;
+
+        if (isString(checkAccess)) {
+            if (user.hasRole(checkAccess)) return true;
+            this.accessDeniedMessage = `User needs the role "${checkAccess}" to access this application.`;
+            return false;
+        } else {
+            const ret = checkAccess(user);
+            if (isBoolean(ret)) return ret;
+            this.accessDeniedMessage = ret.message;
```
Comment (lbwexler) [inline svc/IdentityService.ts:86]: only install a non-empty message

File: svc/IdentityService.ts [hunk 3]
```diff
@@ -69,6 +72,23 @@ export class IdentityService extends HoistService {
             .catchDefault();
     }
 
+    checkAccess(): boolean {
```
Comment (lbwexler) [inline svc/IdentityService.ts:75]: comment that this is for framework use only, or move elsewhere

**Pull request merged**